homeassistant/components/template/alarm_control_panel.py

```diff
@@ -0,0 +1,316 @@
+            if str(temp_ids) != MATCH_ALL:
+                template_entity_ids |= set(temp_ids)
+        else:
+            _LOGGER.warning("No value template - will use optimistic state")
+
+        if not template_entity_ids:
+            template_entity_ids = MATCH_ALL
+
+        alarm_control_panels.append(
+            AlarmControlPanelTemplate(
+                hass,
+                device,
+                friendly_name,
+                state_template,
+                disarm_action,
+                arm_away_action,
+                arm_home_action,
+                arm_night_action,
+                code_arm_required,
+                template_entity_ids,
+            )
+        )
+
+    async_add_entities(alarm_control_panels)
+
+
+class AlarmControlPanelTemplate(AlarmControlPanel):
+    """Representation of a templated Alarm Control Panel."""
+
+    def __init__(
+        self,
+        hass,
+        device_id,
+        friendly_name,
+        state_template,
+        disarm_action,
+        arm_away_action,
+        arm_home_action,
+        arm_night_action,
+        code_arm_required,
+        template_entity_ids,
+    ):
+        """Initialize the panel."""
+        self.hass = hass
+        self.entity_id = async_generate_entity_id(
+            ENTITY_ID_FORMAT, device_id, hass=hass
+        )
+        self._name = friendly_name
+        self._template = state_template
+        self._disarm_script = None
+        self._code_arm_required = code_arm_required
+        if disarm_action is not None:
+            self._disarm_script = Script(hass, disarm_action)
+        self._arm_away_script = None
+        if arm_away_action is not None:
+            self._arm_away_script = Script(hass, arm_away_action)
+        self._arm_home_script = None
+        if arm_home_action is not None:
+            self._arm_home_script = Script(hass, arm_home_action)
+        self._arm_night_script = None
+        if arm_night_action is not None:
+            self._arm_night_script = Script(hass, arm_night_action)
+
+        self._state = None
+        self._entities = template_entity_ids
+
+        if self._template is not None:
+            self._template.hass = self.hass
+
+    @property
+    def name(self):
+        """Return the display name of this alarm control panel."""
+        return self._name
+
+    @property
+    def should_poll(self):
+        """Return the polling state."""
+        return False
+
+    @property
+    def state(self):
+        """Return the state of the device."""
+        return self._state
+
+    @property
+    def supported_features(self) -> int:
+        """Return the list of supported features."""
+        supported_features = (
+            SUPPORT_ALARM_ARM_NIGHT if (self._arm_night_script is not None) else 0
+        )
+        supported_features = (
+            supported_features | SUPPORT_ALARM_ARM_HOME
+            if (self._arm_home_script is not None)
```

Comment:
Please write if checks on multiple lines if it spans more than one line.

Reply:
Made this generally more readable. Ternary -> icky.